File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int insertPos = 0;

        // Step 1: Move all non-zero values forward
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] != 0) {
                nums[insertPos++] = nums[i];
            }
        }

        // Step 2: Fill remaining slots with 0
        while (insertPos < nums.size()) {
            nums[insertPos++] = 0;
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        count = 0
        while i < len(nums):
            if nums[i] == 0:
                count += 1
                nums.remove(nums[i])
            i +=1
            if count!= 0:
                nums.append(0)
                count -= 1
